File: Frontend/src/app/Models/Habitacion.ts
import { CuentaHabitacion } from './CuentaHabitacion';
import { TipoHabitacion } from './TipoHabitacion';

export interface Habitacion {
  idHabitacion?: number;
  numero: string;
  precioPorNoche: number;
  estado: string;
  capacidad: number;
  descripcion: string;
  tipoHabitacion?: TipoHabitacion;
  // Cuenta Habitacion Eliminada
}

File: Backend/src/main/java/com/example/lumaresort/entities/Habitacion.java
package com.example.lumaresort.entities;

import jakarta.persistence.*;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@ToString(exclude = {"tipoHabitacion"})
public class Habitacion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idHabitacion;

    private String numero;
    private float precioPorNoche;
    private String estado;
    private Integer capacidad;
    private String descripcion;

    @ManyToOne
    @JoinColumn(name = "idTipoHabitacion")
    private TipoHabitacion tipoHabitacion;
}
